src/modsim2/similarity/metrics/pad.py

```diff
@@ -81,34 +70,59 @@ def __train_test_split(
             train_B: records in dataset B to be used for training
             test_B: records in dataset B to be used for testing (final evaluation)
         """
+
+        if self.train_balance == "equal":
+            # If the data_A and data_B have different numbers of records then drop
+            # records from the larger of the two datasets so that A & B are of
+            # equal size
+            if data_A.shape[0] > data_B.shape[0]:
+                data_A = self._reduce_size_array(data_A, data_B.shape[0])
+            elif data_B.shape[0] > data_A.shape[0]:
+                data_B = self._reduce_size_array(data_B, data_A.shape[0])
+        elif self.train_balance != "ratio":
```

Comment:
e.g. (though could use refinement!)
```
       if self.train_balance == "equal":
            # If the data_A and data_B have different numbers of records then drop
            # records from the larger of the two datasets so that A & B are of
            # equal size
            if data_A.shape[0] > data_B.shape[0]:
                data_A = self._reduce_size_array(data_A, data_B.shape[0])
            elif data_B.shape[0] > data_A.shape[0]:
                data_B = self._reduce_size_array(data_B, data_A.shape[0])

                test_size_A = int(np.round(data_A.shape[0] * self.test_proportion))
                # use same size for both, as equal size
                test_size_B = test_size_A

                # Split A&B into train and test
                train_A, test_A = train_test_split(data_A, test_size=test_size_A, shuffle=True)
                train_B, test_B = train_test_split(data_B, test_size=test_size_B, shuffle=True)

                return train_A, test_A, train_B, test_B

        elif self.train_balance == "ratio":
            test_size_A = int(np.round(data_A.shape[0] * self.test_proportion))
            test_size_B = int(np.round(data_B.shape[0] * self.test_proportion))

            # Split A&B into train and test
            train_A, test_A = train_test_split(data_A, test_size=test_size_A, shuffle=True)
            train_B, test_B = train_test_split(data_B, test_size=test_size_B, shuffle=True)

            # test_A and test_B must have the same number of records for this metric
            # If one is greater than the other (which may occur when the ratio between A&B
            # has been preserved in the training data) then the excess records will
            # be dropped at random
            if test_A.shape[0] > test_B.shape[0]:
                test_A = self._reduce_size_array(test_A, test_B.shape[0])
            elif test_B.shape[0] > test_A.shape[0]:
                test_B = self._reduce_size_array(test_B, test_A.shape[0])

            return train_A, test_A, train_B, test_B
        else:
            raise ValueError(
                "An invalid train_balance value has been provided", self.train_balance
            )
```

@philswatton I think this method has overlap with a discussion on preserving ratio we had in standup? any changes you make may render the above irrelevant.

Reply:
Yep, I've made changes on this already - will push up tomorrow morning once I've finished checking that I've updated the docstrings correctly!